### DISCUSSION TITLE
Can I use database transactions?

### DISCUSSION THREAD

**Discussion by kjurenec** [keystonejs/keystone#7642] [Questions] [answered]
Does keystore support transactions? I know that prisma does but when i try something like this (item name is unique) one item is created and then the rest of it fails.
```gql
mutation {
  createCategory(
    data: { name: "C1", items: { create: [{ name: "I1" }, { name: "I1" }] } }
  ) {
    id
    name
    items {
      id
      name
    }
  }
}
```

**Comment by dcousens** [comment, marked as answer]
Hi @kjurenec,

There is no support for database transactions at this time,  which is unfortunate.
We aren't unaware of the problem, and it is something we are trying to integrate into existing functionality, but for now, it is not supported. 

The problems aren't new to us, but different solutions have different trade-offs which we are always trying to balance with productivity and availability.

#### Related
- https://github.com/keystonejs/keystone/pull/207
- https://github.com/keystonejs/keystone/issues/211
- https://github.com/keystonejs/keystone/issues/534
- https://github.com/keystonejs/keystone/pull/535
- https://github.com/keystonejs/keystone/discussions/7642
- https://github.com/keystonejs/keystone/issues/7217
- https://github.com/keystonejs/keystone/discussions/8084

**Reply by dcousens** [reply]
Transactions have now been added in https://github.com/keystonejs/keystone/pull/9122

**Comment by korzewski** [comment]
Hey, I'm using "@keystone-6/core: 3.1.1" and issue with transactions still occurs.

I wonder if this was solved at any further version, so if I upgrade to the newest (5.7.2) this would be fixed already?
